### PR TITLE
User profile update

### PR DESCRIPTION
**Pull request by flierc-csp**
Profile update and profile image upload

### CODE REVIEW

File: docker/api/src/Domain/User/User.php [hunk 1]
```diff
@@ -190,9 +193,14 @@ public function verifyPassword(string $password): bool{
         return password_verify($password, $this->password_hash);
     }
 
-    #[Pure] #[ArrayShape(['userId' => "int", 'firstName' => "string", 'lastName' => "string", 'email' => "string", 'balance' => "string", 'shoppingCart' => 'array', 'address' => "string",'city' => "string",'state' => "string",'zip' => "string",'country' => "string"])]
+    #[Pure] #[ArrayShape(['userId' => "int", 'firstName' => "string", 'lastName' => "string", 'email' => "string", 'balance' => "string", 'shoppingCart' => 'array', 'address' => "string",'city' => "string",'state' => "string",'zip' => "string",'country' => "string", 'profileImage' => 'string'])]
     public function jsonSerialize(): array{
 
+        //Profile Image
+        $userId = $_COOKIE['freshMartUserId'];
```
Comment (noko12312) [inline docker/api/src/Domain/User/User.php]: On the Checkout page this line is causing the following error response:
`"ERROR: Undefined array key \"freshMartUserId\" on line 200 in file \/var\/www\/html\/api\/src\/Domain\/User\/User.php."`
Can you check if you are getting this as well?

**Review by noko12312 — COMMENTED** [review]
When trying to update profile picture I am getting the following error in the response:
`"Upload target path is not writable"`

File: docker/api/src/Domain/User/User.php (hunk 1 above)
Comment (flierc-csp) [reply docker/api/src/Domain/User/User.php]: I didn't consider  non-authenticated requests when I wrote that.  It now checks to make sure the cookie is set first.

**Comment by thebaker10** [comment]
I am getting this error when I try to login
`{
    "statusCode": 500,
    "error": {
        "type": "SERVER_ERROR",
        "description": "ERROR: hash(): Passing null to parameter #2 ($data) of type string is deprecated on line 254 in file \/var\/www\/html\/api\/src\/Domain\/User\/User.php."
    }
}`

**Comment by noko12312** [comment]
@thebaker10 I am getting the same error on login

@flierc-csp Is this just because we do not have the COOKIE_SALT environment variable set on our end?

**Review by noko12312 — APPROVED** [review]

**Pull request merged**